react-native-aztec/ios/RNTAztecView/RCTAztecView.swift

```diff
@@ -175,26 +175,34 @@ class RCTAztecView: Aztec.TextView {
     }
     
     // MARK: - Paste handling
-    
-    private func html(from pasteboard: UIPasteboard) -> String? {
-        guard let data = pasteboard.data(forPasteboardType: kUTTypeHTML as String) else {
-            return nil
+    private func readHTML(from pasteboard: UIPasteboard) -> String? {
+        if let data = pasteboard.data(forPasteboardType: kUTTypeHTML as String) {
+            return String(data: data, encoding: .utf8)
         }
-        
-        return String(data: data, encoding: .utf8)
+
+        if let data = pasteboard.data(forPasteboardType: kUTTypeRTF as String),
+            let attributedString = try? NSAttributedString(data: data, options: [.documentType: DocumentType.rtf], documentAttributes: nil),
+            let storage = self.textStorage as? TextStorage {
+                return  storage.getHTML(from: attributedString)
+        }
+
+        return nil
     }
     
-    private func text(from pasteboard: UIPasteboard) -> String? {
-        guard let data = pasteboard.data(forPasteboardType: kUTTypePlainText as String) else {
-            return nil
+    private func readText(from pasteboard: UIPasteboard) -> String? {
+        if let data = pasteboard.data(forPasteboardType: kUTTypeUTF8PlainText as String) {
+            return String(data: data, encoding: .utf8)
         }
-        
-        return String(data: data, encoding: .utf8)
-    }
 
-    private func cleanHTML() -> String {
-        let html = getHTML(prettify: false)
-        return html
+        if let data = pasteboard.data(forPasteboardType: kUTTypePlainText as String) {
+            return String(data: data, encoding: .ascii)
+        }
+
+        if let data = pasteboard.data(forPasteboardType: kUTTypeText as String) {
+            return String(data: data, encoding: .ascii)
```

Comment:
Why `.ascii` and not `.utf8`?

Reply:
when the UTI is TypeText you don't have any guarantees of the encoding. So I was trying to put the lowest denominator there. It's a pity the platform doesn't have an auto-discover mode for this.

Reply:
I think my slack comment got lost in the threads, so replying here. Are you checking for `kUTTypePlainText` and `kUTTypeText` for a reason?

https://developer.apple.com/documentation/uikit/uipasteboard/1622091-strings makes me think that `kUTTypeUTF8PlainText` is the only one relevant